src/numpy.jl

```diff
@@ -154,13 +156,14 @@ npy_type(::Type{Int32}) = NPY_INT
 npy_type(::Type{UInt32}) = NPY_UINT
 npy_type(::Type{Int64}) = NPY_LONGLONG
 npy_type(::Type{UInt64}) = NPY_ULONGLONG
+npy_type(::Type{Float16}) = NPY_HALF
 npy_type(::Type{Float32}) = NPY_FLOAT
 npy_type(::Type{Float64}) = NPY_DOUBLE
 npy_type(::Type{Complex64}) = NPY_CFLOAT
 npy_type(::Type{Complex128}) = NPY_CDOUBLE
 npy_type(::Type{PyPtr}) = NPY_OBJECT
 
-typealias NPY_TYPES Union{Bool,Int8,UInt8,Int16,UInt16,Int32,UInt32,Int64,UInt64,Float32,Float64,Complex64,Complex128,PyPtr}
+typealias NPY_TYPES Union{Bool,Int8,UInt8,Int16,UInt16,Int32,UInt32,Int64,UInt64,Float16,Float32,Float64,Complex64,Complex128,PyPtr}
 
 # conversions from __array_interface__ type strings to supported Julia types
 const npy_typestrs = Dict( "b1"=>Bool,
```

Comment:
You also need an entry `"f2"=>Float16` in `npy_typestrs`, right (in order for conversions in the other direction to work)?